src/components/hooks/use-browser-resize.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react'
+import { isBrowser } from 'common/utility'
+import { size } from 'themes/device'
+
+export const useBrowserResize = () => {
+    const [is_mobile, setMobile] = useState(false)
+    const [width, setWidth] = useState(isBrowser() ? window.innerWidth : 0)
```

Comment:
This is great 👍 We're using this in a couple of places. We can replace them with this hook in the future.